Samples/WindowPlacement/cpp/inc/WindowActions.h

```diff
@@ -0,0 +1,386 @@
+#pragma once
+// Note: Included by User32Utils.h, if USE_WINDOW_ACTION_APIS is defined.
+
+//
+// Note: These APIs are only available on newer OS builds. Apps using these
+// APIs that need to run on older releases should handle a fallback when the
+// APIs fail.
+//
+
+// ========================================================================
+
+//
+// CWindowAction
+//
+// Helper class to set fields in a WINDOW_ACTION and call ApplyWindowAction.
+//
+// A Window Action describes changes to make to a top level window. Moving,
+// sizing, activating, maximizing, etc.
+//
+// The 'kinds' field of the action is flags that describe the changes to make.
+// Some have no additional payload, like WAK_ACTIVATE, and others like
+// WAK_POSITION have a corresponding field containing the value.
+//
+// The 'modifiers' field is also flags, some of which have their own field. The
+// modifiers change the behavior of one or more of the kinds in some way. For
+// example, the modifier WAM_FRAME_BOUNDS changes how the provided rect
+// (position/size) are interpretted.
+//
+class CWindowAction : public WINDOW_ACTION
+{
+public:
+    CWindowAction()
+    {
+        RtlZeroMemory(this, sizeof(this));
```

Comment:
Why do we need to zero this memory? Doesn't this happen automatically for structs?